src/search/engines/_sedefaults/prepare.go

```diff
@@ -9,10 +9,9 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
-func Prepare(seName engines.Name, options *engines.Options, settings *config.Settings, support *engines.SupportedSettings, info *engines.Info, ctx *context.Context) error {
-	if ctx == nil {
-		*ctx = context.Background()
-	}
+// sending options and settings as pointers since they are modified
+func Prepare(ctx context.Context, info engines.Info, support engines.SupportedSettings, options *engines.Options, settings *config.Settings) (context.Context, error) {
```

Comment:
Why is `ctx` even passed to prepare? Can `ctx` even ever be `nil` since its not a pointer?

Reply:
context.Context is na interface, it can be set to `nil`